cicecore/cicedynB/general/ice_init.F90

```diff
@@ -2868,39 +2884,55 @@ subroutine set_state_var (nx_block, ny_block, &
          ! ice concentration/thickness
          !---------------------------------------------------------
 
-         if (trim(ice_data_type) == 'box2001'    .or. &
-             trim(ice_data_type) == 'smallblock' .or. &
-             trim(ice_data_type) == 'channel'    .or. &
-             trim(ice_data_type) == 'bigblock'   .or. &
-             trim(ice_data_type) == 'blockep5'   .or. &
-             trim(ice_data_type) == 'uniformp5'  .or. &
-             trim(ice_data_type) == 'gauss') then
+         if (trim(ice_data_conc) == 'p5') then
 
             hbar = c2  ! initial ice thickness
             do n = 1, ncat
                hinit(n) = c0
                ainit(n) = c0
                if (hbar > hin_max(n-1) .and. hbar < hin_max(n)) then
                   hinit(n) = hbar
-                  ainit(n) = p5 !echmod symm
+                  ainit(n) = p5
                endif
             enddo
 
-         elseif (trim(ice_data_type) == 'boxslotcyl' .or. &
-                 trim(ice_data_type) == 'medblocke'  .or. &
-                 trim(ice_data_type) == 'blocke') then
+         elseif (trim(ice_data_conc) == 'p8') then
+
+            hbar = c2  ! initial ice thickness
+            do n = 1, ncat
+               hinit(n) = c0
+               ainit(n) = c0
+               if (hbar > hin_max(n-1) .and. hbar < hin_max(n)) then
+                  hinit(n) = hbar
+                  ainit(n) = 0.8_dbl_kind
```

Comment:
Instead of repeating this whole block of code for each ice_data_conc option, I'd set local parameters hbar and abar in conditionals then load them into hinit and ainit afterward.  That just a style thing, which makes it obvious what the essential differences are.  The 'default' option is different, but see comments below.

Reply:
This is done.